Adjust RecomputeGeneral class ...

jasondhales opened the pull request:
Adjust Recompute classes.  This will enable use of temperature-dependent moduli.  No tests were changed as this change enables future development but does not alter end behavior.  #5481

sapitts commented on modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C:
```diff
@@ -38,8 +37,8 @@ ComputeReturnMappingStress::computeQpStress()
   // compute the stress increment (as a tensor) and the inelastic strain increment (tensor) required to return to the yield surface
```
Would you mind updating this comment to clarify that the whole stress, not the stress increment, is returned please?

jasondhales replied:
Done.

jasondhales committed:
Rework the Recompute classes

This is done so that the new stress is computed based on the current
elastic strain.  This allows temperature-dependent moduli to affect
the stress properly.

Genuflected at the Throne of Whitespace.

ref #5481

jwpeterson commented on modules/tensor_mechanics/src/materials/RecomputeRadialReturnIsotropicPlasticity.C:
```diff
@@ -6,10 +6,12 @@
 /****************************************************************/
 #include "RecomputeRadialReturnIsotropicPlasticity.h"
 
+#include "Function.h"
```
Did this include need to be added?  I didn't see any other changes in the file that would appear to require it.

jasondhales replied:
Won't compile without it.

jwpeterson commented on modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C:
```diff
@@ -19,24 +19,24 @@ InputParameters validParams<RecomputeRadialReturnStressIncrement>()
-    _absolute_tolerance(parameters.get<Real>("absolute_tolerance"))
+RecomputeRadialReturn::RecomputeRadialReturn(const InputParameters & parameters) :
+  RecomputeGeneralReturn(parameters),
+  _output_iteration_info(getParam<bool>("output_iteration_info")),
```
I think this indent needs to stay at 4 spaces, or @andrsd gets mad.

jasondhales replied:
OK.  I'll change it to four spaces.  We all realize this is silly, right?  Why are four spaces in an initialization list better than two (used everywhere else)?

jwpeterson commented on modules/tensor_mechanics/src/materials/RecomputeGeneralReturn.C:
```diff
@@ -0,0 +1,47 @@
+
+RecomputeGeneralReturn::RecomputeGeneralReturn(const InputParameters & parameters) :
+  Material(parameters),
+  _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : "" ),
```
Also 4 spaces here.

permcody commented:
👍
